tests/create_test.go

```diff
@@ -548,6 +548,35 @@ func TestFirstOrCreateRowsAffected(t *testing.T) {
 	}
 }
 
+func TestCreateWithAutoIncrementCompositeKey(t *testing.T) {
+	type CompositeKeyProduct struct {
+		ProductID    int `gorm:"primaryKey;autoIncrement:true;"` // primary key
+		LanguageCode int `gorm:"primaryKey;"`                    // primary key
+		Code         string
+		Name         string
+	}
+
+	if err := DB.AutoMigrate(&CompositeKeyProduct{}); err != nil {
+		t.Fatalf("failed to migrate, got error %v", err)
+	}
+
+	prod := &CompositeKeyProduct{
+		LanguageCode: 56,
+		Code:         "Code56",
+		Name:         "ProductName56",
+	}
+	if err := DB.Create(&prod).Error; err != nil {
+		t.Fatalf("failed to create, got error %v", err)
+	}
+
+	newProd := &CompositeKeyProduct{}
+	if err := DB.First(&newProd).Error; err != nil {
```

Comment:
This is not rigorous here. When I run the test repeatedly, the inserted record product id is self-increasing, so it will fail.
I suggest to do `DropTable` before `AutoMigrate`.

```golang
if err := DB.Migrator().DropTable(&CompositeKeyProduct{}); err != nil {
	t.Fatalf("failed to migrate, got error %v", err)
}
```